libs/sdk-core/src/models.rs

```diff
@@ -1215,6 +1236,58 @@ impl FromStr for BuyBitcoinProvider {
     }
 }
 
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct PaymentPath {
+    pub edges: Vec<PaymentPathEdge>,
+}
+
+impl PaymentPath {
+    pub fn final_hop_amount(&self, first_hop_amount_msat: u64) -> u64 {
+        let mut max_to_send = first_hop_amount_msat;
+        for h in self.edges.iter().skip(1) {
+            max_to_send = h.amount_to_forward(max_to_send);
+        }
+        max_to_send
+    }
+
+    pub fn first_hop_amount(&self, final_hop_amount_msat: u64) -> u64 {
+        let mut first_hop_amount = final_hop_amount_msat;
+        for h in self.edges.iter().skip(1).rev() {
+            first_hop_amount = h.amount_from_forward(first_hop_amount);
+        }
+        first_hop_amount
+    }
+}
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct PaymentPathEdge {
+    pub node_id: Vec<u8>,
+    pub short_channel_id: String,
+    pub channel_delay: u64,
+    pub base_fee_msat: u32,
+    pub fee_per_millionth: u32,
+}
+
+impl PaymentPathEdge {
+    pub(crate) fn amount_to_forward(&self, in_amount_msat: u64) -> u64 {
+        let forward = (in_amount_msat - self.base_fee_msat as u64) as f64
+            / (1f64 + self.fee_per_millionth as f64 / 1000000f64);
```

Comment:
`1000000f64` would be better readable as `1_000_000_f64`

Reply:
Fixed